### ISSUE TITLE
[2.16.0] Computed filter regression

### ISSUE THREAD

**Issue by ilucin** [emberjs/ember.js#15843]
There's a regression bug in computed.filter not observing property changes properly -> in 2.16 release. 

Paste this into the console:
```
Obj = Ember.Object.extend({
  filteredItems: Ember.computed.filter('items.@each.{prop}', function(item) {
    return item.get('prop') === false;
  })
});

obj = Obj.create();
obj.set('items', [Ember.Object.create({prop: false})]);
console.log('Expected: 1 Actual: ', obj.get('filteredItems.length'));
obj.get('items')[0].set('prop', true);
console.log('Expected: 0 Actual: ', obj.get('filteredItems.length'));
```

This is the third core regression bug (#15836, #15733) I found in 2.16 and this one is pretty major. What's happening here, are there any more? Shall I PR failing tests in core_object module? @rwjblue

**Comment by rwjblue** [comment]
Yes, a failing test PR would be helpful. I'm not sure of the root cause without digging yet...

**Comment by ilucin** [comment]
Here it is. I've scanned the 2-15.3 -> 2.16.0 changes and I can see that there has been some ember-core updating going on but I can't find the source of these regressions.

**Comment by Serabe** [comment]
Fixed in #15855
